### DISCUSSION TITLE
Not escaping frontmatter values (or a work around)

### DISCUSSION THREAD

**Discussion by eelcoj** [bridgetownrb/bridgetown#726] [Q&A]
I use a (Netlify) function that creates images on the fly. Great way to make og:images stand out more. 

The problem I'm facing is around xml escaping that (rightly so) in place for frontmatter values.

So the following line:

```
image: https://spinal-og-images.netlify.app/templates/framework-template/images/open-graph.jpg?name=forestry&type=alternative&heading=Alternative%20to%20Forestry
```

turns into:

```
https://spinal-og-images.netlify.app/templates/framework-template/images/open-graph.jpg?name=forestry&amp;type=alternative&amp;heading=Alternative%20to%20Forestry
```

Notice the `&amp;`. 

I've tried the parsing the frontmatter using Ruby, but that doesn't help as the escaping happens later. Any ideas or workarounds known for this?

**Comment by neilvanbeinum** [comment]
I'm not too clear on the sequence of events in your implementation and it's probably because I'm not too familiar with this part of Bridgetown. How are you calling the Netlify function? Are you calling it within Ruby frontmatter?

**Reply by eelcoj** [reply]
I tried Ruby frontmatter, but the escaping happens regardless. Ideally want vanilla frontmatter. 

There's no Netlify function involved that concerns BT. All it needs is a link, from Frontmatter, like the following:

```yml
---
image: https://spinal-og-images.netlify.app/templates/framework-template/images/open-graph.jpg?name=forestry&type=alternative&heading=Alternative%20to%20Forestry
---
```

Thinking of "monkey-patching" the method in some way, but that's new BT territory for me.

**Comment by jaredcwhite** [comment]
I would be very surprised if the front matter value itself is being escaped in any way. Can you describe how you're using it within a template? There's probably a way to output the raw string.

**Reply by eelcoj** [reply]
Spent some time on this. I've went back to the basics and set up a new BT site. I now knew:

- it was using [Bridgetown seo tag plugin](https://github.com/bridgetownrb/bridgetown-seo-tag)
- noticed it was working as expected
- also noticed I was using an older version; so updated
- now everything is working without problems
- dove into the internals of the plugin and checked the last commits, but can't find anything different

Bit stumped, but it's working so ¯\_(ツ)_/¯
